src/nodejs/neo-rpc/v2.9.0/client/cli/query.js

```diff
@@ -54,10 +57,20 @@ if (program.debug) {
 }
 
 if (!program.node) {
+  // get a node from the list and try it
+  let net = netUtil.resolveNetworkId(program.Net)
+
+  nodes = cfg.getNodes(net)
+
   let options = {
-    net: program.Net,
+    net: net,
+    order: 'asc',
+    nodes: nodes
   }
-  netUtil.getNodesByPing(options).then(rankedNodes => {
+
+  if (defly) dbg.logDeep('config nodes: ', nodes)
+
+  getNodesBy.tallest(options).then(rankedNodes => {
```

Comment:
As this has a disclaimer saying "caution: can cause a lot of traffic"
isn't it a bad idea to use it internally?

Reply:
Not sure yet, I've been mulling this one over. I don't think it's that big of a deal, but I do feel that users should be aware. I'm taking a note to add this advisory anywhere this code is used. I'm also considering adding an option to disable dynamic node selection by default so that traffic issue disappears. Good catch and thank you.